### DISCUSSION TITLE
Rotation of 45 degree for Insert Text in PDF

### DISCUSSION THREAD

**Discussion by khetaniarpan** [pymupdf/PyMuPDF#2722] [Q&A] [answered]
I am trying to insert text for landscape and potrait mode pages , but i noticed,  we cannot do rotation on 45 degrees. Do we have any option to do. I noticed your previous codes , and tried this scenario , but no way to rotate at an angle of 45.

Can you please help
p1 = fitz.Point(100, 100)
p9 = fitz.Point(900, 900)
page.insert_text(p1 * page.derotation_matrix, "test_100", fontsize=50, rotate=page.rotation)
page.insert_text(p9 * page.derotation_matrix, "Preliminary Report", fontsize=100, rotate=page.rotation)

![image](https://github.com/pymupdf/PyMuPDF/assets/132350977/bf4cd829-bbf9-4ef3-ba3f-74099fdcbf25)

**Comment by JorjMcKie** [comment]
This is a typical "Discussions" post. Let me first convert it.

**Reply by JorjMcKie** [reply, marked as answer]
Simply use the `morph` parameter: you must provide `morph=(point, matrix)`, where `point`is a `fitz.Point` to be used as a fixpoint and `matrix` a `fitz.Matrix`. This can be any any matrix with `matrix.e = matrix.f = 0`.
So to show inserted text rotated by 45° around the insertion point do `page.insert_text(point, text, morph=(point, fitz.Matrix(45))`.

**Reply by khetaniarpan** [reply]
Thank you for your quick response , and helping me out. It worked perfectly fine :)

**Reply by JorjMcKie** [reply]
The morph parameter is also supported in a number of other places like `insert_textbox()` and more. Check out the documentation!
